# How to calculate Sharpness from .wav file (steady signal)

MOSQITO is a unified and modular development framework of key sound quality metrics favoring reproducible science and efficient shared scripting among engineers, teachers and researchers community.

This tutorial explains how to calculate the acoustic sharpness of a steady signal from its loudness and specific loudness values (Zwicker method) as described in DIN 45692_2009E.

In [1]:
# Add MOSQITO to the Python path
import sys
sys.path.append('..')

# Import useful packages
import numpy as np

# Import MOSQITO module
from mosqito.functions.shared.load import load
from mosqito.functions.sharpness.comp_sharpness import comp_sharpness

For this tutorial is used a broad-band test noise from DIN 45692_2009E with a center frequency of 2500Hz.
The signal is loaded with the "load" function.

In [2]:
# Load signal 
sig, fs = load(True,r"..\mosqito\validations\sharpness\data\Check_signals_DIN_45692_(Schaerfe)\Broadband_noise(fo=10kHz_fu=variabel)\LC2500.WAV")

Then the sharpness is calculated with the 'comp_sharpness' function which automatically does the loudness calculation needed. The method chosen to do the calculation is given as a parameter (the parameter 'all' launches all the different calculation methods).


In [4]:
sharpness = comp_sharpness(True, sig, fs, method='din')

DIN sharpness: 3.6876236544360963 acum


The reference value given in DIN 45692 for this signal is 3.69, therefore the Mosqito implementation of the 'din' method fills the compliance with the standard for this signal (5% or 0.05 tolerance).

Mosqito offers several other calculation methods for the sharpness, which can be computed by changing the 'method' parameter :


In [5]:
# Comparison with other methods
sharpness = comp_sharpness(True, sig, fs, method='bismarck')
sharpness = comp_sharpness(True, sig, fs, method='fastl')
sharpness = comp_sharpness(True, sig, fs, method='aures')

Bismarck sharpness: 3.4708017062520145 acum
Fastl sharpness: 3.321735750475353 acum
Aures sharpness: 5.083057395100374 acum


The values slightly differ due to the weighting functions differences, except for Aures calculation which differ more (the weighting function takes into account the global loudness value).